src/ray/gcs/gcs_client/gcs_client.cc

```diff
@@ -78,6 +78,24 @@ void GcsSubscriberClient::PubsubCommandBatch(
       });
 }
 
+// Waits for the channel to be ready.
+// The wait time takes a fraction of the total timeout of an RPC, so we limit out wait
+// time to min(rpc total timeout, configured connect timeout).
+Status WaitForChannelReady(grpc::Channel &channel, int64_t rpc_timeout_ms) {
+  auto timeout_duration = std::chrono::duration_cast<std::chrono::milliseconds>(
+      std::chrono::seconds(::RayConfig::instance().gcs_rpc_server_connect_timeout_s()));
```

Comment:
Actually, should we just make it 1 second? 

1. It is strictly better than status quo (status quo is failing immediately)
2. Looks like many tests require 1 second, and adding 5 seconds will add serious delay in many RPC path

Reply:
made 1 second for the initial connect, this should suffice for the unit tests.